modules/puppet_proxy/puppet_class.rb

```diff
@@ -37,11 +37,25 @@ def name
   end
 
   attr_reader :params
-
-  private
   attr_reader :klass
 
   def has_module?(klass)
     !!klass.index("::")
   end
+
+  def to_json(*a)
+    {
+        'json_class' => self.class.name,
```

Comment:
I assume this triggers the deserialisation process to call json_create.  I just tried on an EL6 installation with json 1.5.5 (note, that's not easy to change) and I think we're just getting hashes rather than classes:

    Failed to show puppet classes: undefined method `name' for #<Hash:0x7f58598147d8>

(I think that's in the puppet_api.)

Does this require a min version of json to work?  Could we transform the hashes into objects in our deserialisation method instead of relying on json to do it?